src/dotnet-install.ps1

```diff
@@ -545,60 +545,84 @@ function Get-LegacyDownload-Link([string]$AzureFeed, [string]$SpecificVersion, [
 function Get-Product-Version([string]$AzureFeed, [string]$SpecificVersion, [string]$PackageDownloadLink) {
     Say-Invocation $MyInvocation
 
+    $ProductVersionTxtURLs = (Get-Product-Version-Url $AzureFeed $SpecificVersion $PackageDownloadLink $true),(Get-Product-Version-Url $AzureFeed $SpecificVersion $PackageDownloadLink $false)
+    
+    Foreach ($ProductVersionTxtURL in $ProductVersionTxtURLs) {
```

Comment:
honestly it took me a while to understand the logic :) or recall it
i think 2 comments (or even verbose logs) will be very helpful:

at line 548: try to read productversion.txt file at package location
at line 573: try to get product version from download link 

also I regret for putting download link check to `ProductVersionFromDownloadLink` function 
I believe to make logic of this method complete it would be good to move the following code to this method: 
```
    if ([string]::IsNullOrEmpty($PackageDownloadLink))
    {
        Say-Verbose "Using the default value '$SpecificVersion' as the product version."
        return $SpecificVersion
    }

```